### PR TITLE
pre-process data

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR pre-processes the data and aggregates where needed.

### CODE REVIEW

File: 0.Preprocess_Data/README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
-# Parquet File Merge
+# Preprocess Data
 
-This processing run was broken up into two separate runs as there was as power failure during the first half.
+Here we preprocess the data to be used in the model. The data is stored in the `data` folder.
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/README.md:3]:
```suggestion
Here we preprocess the data to be used in the model. 
The data is stored in the `data` folder.
```
One line per sentence.

File: 0.Preprocess_Data/README.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+The data is preprocessed in the following steps:
+
+1. Morphology preprocessing
+    Here we clean the morphoplogical data and metadata.
+2. nELISA (Secretome) preprocessing
+    Here we clean the nELISA data and metadata.
+3. Aggregation and Merging
+    Here we aggregate the moprhology data to the well level and merge it with the nELISA data.
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/README.md:11]:
```suggestion
    Here we aggregate the morphology data to the well level and merge it with the nELISA data.
```

File: 0.Preprocess_Data/pre-processing.sh [hunk 3]
```diff
@@ -15,11 +16,11 @@ cell_types=( PBMC SHSY5Y )
 aggragates=( True False )
 nomics=( True False )
 
-for cell_type in $cell_types; do
+for cell_type in ${cell_types[@]}; do
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/pre-processing.sh:19]: Just curious, what does this `@` symbol do here? Is this related to papermill?

File: 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py [hunk 4]
```diff
@@ -239,40 +239,75 @@
...
+    "Flagellin_1.000_0_DMSO_0.025_%",
+    "Flagellin_1.000_ug_per_ml_DMSO_0.025_%",
+    regex=False,
+)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_0.100_ug_per_ml_DMSO_0.000_%",
+    "Flagellin_0.100_ug_per_ml_DMSO_0.025_%",
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py:272]: A bit confused here, it looks to me here that you are replacing the treatment with 0% DMSO to 0.025% DMSO. Is this correct?

File: 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py [hunk 5]
```diff
@@ -239,40 +239,75 @@
...
+    regex=False,
+)
 
-# In[19]:
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_ug_per_ml_DMSO_0.000_%",
+    "Flagellin_1.000_ug_per_ml_DMSO_0.025_%",
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py:280]: Same here.

File: 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py [hunk 6]
```diff
@@ -239,40 +239,75 @@
...
+    if feature_df[column].dtype == "object":
+        feature_df[column] = feature_df[column].astype(str)
 
-# Define inputs
-feature_file = pathlib.Path(f"../data/{cell_type}_sc.parquet")
-feature_df = pd.read_parquet(feature_file)
-# quick check for site
-print(feature_df["Metadata_Site"])
+
+# In[ ]:
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py:309]: Looks like this code cell wasn't run. Recommend making sure that all cells are run sequentially.

File: 0.Preprocess_Data/scripts/2.preprocessing_nELISA_data.py [hunk 7]
```diff
@@ -272,14 +272,44 @@ def perform_replacements(text: str) -> str:
...
+].str.replace("media_ctr_0.0_0_Media_ctr_0.0_0", "Media")
+
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_0.0_0_Media_0.0_0", "Media")
+
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_0.100_ug_per_ml_DMSO_0.0_%", "Flagellin_0.100_ug_per_ml_DMSO_0.025_%"
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/scripts/2.preprocessing_nELISA_data.py:286]: Same comment as before.

File: 0.Preprocess_Data/scripts/2.preprocessing_nELISA_data.py [hunk 8]
```diff
@@ -272,14 +272,44 @@ def perform_replacements(text: str) -> str:
...
+)
+
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_ug_per_ml_DMSO_0.0_%", "Flagellin_1.000_ug_per_ml_DMSO_0.025_%"
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/scripts/2.preprocessing_nELISA_data.py:292]: Same comment as before.

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! I left a few clarifying comments, feel free to merge when addressed! :tada:

File: 0.Preprocess_Data/pre-processing.sh (hunk 3 above)
Comment (MikeLippincott) [reply 0.Preprocess_Data/pre-processing.sh:19]: No, these specify to loop through the entire array

File: 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py (hunk 4 above)
Comment (MikeLippincott) [reply 0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py:272]: Yes, correct there was an error in the metadata
